packages/flutter/lib/src/widgets/magnifier.dart

```diff
@@ -0,0 +1,545 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+import 'dart:ui';
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'container.dart';
+import 'framework.dart';
+import 'inherited_theme.dart';
+import 'navigator.dart';
+import 'overlay.dart';
+
+/// [MagnifierController]'s main benefit over holding a raw [OverlayEntry] is that
+/// [MagnifierController] will handle logic around waiting for a magnifier to animate in or out.
+///
+/// If a magnifier chooses to have an entry / exit animation, it should provide the animation
+/// controller to [MagnifierController.animationController]. [MagnifierController] will then drive
+/// the [AnimationController] and wait for it to be complete before removing it from the
+/// [Overlay].
+///
+/// To check the status of the magnifier, see [MagnifierController.shown].
+// TODO(antholeole): This whole paradigm can be removed once portals
+// lands - then the magnifier can be controlled though a widget in the tree.
+// https://github.com/flutter/flutter/pull/105335
+class MagnifierController {
+  /// If there is no in / out animation for the magnifier, [animationController] should be left
+  /// null.
+  MagnifierController({this.animationController}) {
+    animationController?.value = 0;
+  }
+
+  /// The controller that will be driven in / out when show / hide is triggered,
+  /// respectively.
+  AnimationController? animationController;
+
+  /// The magnifier's [OverlayEntry], if currently in the overlay.
+  ///
+  /// This is public in case other overlay entries need to be positioned
+  /// above or below this [overlayEntry]. Anything in the paint order after
+  /// the [RawMagnifier] will not be displayed in the magnifier; this means that if it
+  /// is desired for an overlay entry to be displayed in the magnifier,
+  /// it _must_ be positioned below the magnifier.
+  ///
+  /// {@tool snippet}
+  /// ```dart
+  /// void magnifierShowExample(BuildContext context) {
```

Comment:
Nice example. Might be easier to read if `displayInMagnifierEvenThoughPlacedBeforeChronologically` was just `displayInMagnifier` 😀

Reply:
I was definitely trigger happy with the names😅